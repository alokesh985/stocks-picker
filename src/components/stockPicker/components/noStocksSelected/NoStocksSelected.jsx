import React, { memo } from 'react';

// Styles
import styles from "./noStocksSelected.module.scss";

const NoStocksSelected = () => (
  <div className={styles.container}>
    <h2>No Stocks Selected!</h2>
    <h2>Please search for a stock on the search bar above</h2>
  </div>
);

export default memo(NoStocksSelected);
