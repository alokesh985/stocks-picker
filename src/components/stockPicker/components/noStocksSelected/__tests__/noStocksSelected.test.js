import React from "react";

// Utils
import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";

// Components
import NoStocksSelected from "../NoStocksSelected";

describe("Testing the NoStocksSelected component", () => {
  test("Should show the correct message", () => {
    render(<NoStocksSelected />);
    const line1 = screen.getByText("No Stocks Selected!");
    const line2 = screen.getByText(
      "Please search for a stock on the search bar above"
    );
    expect(line1).toBeInTheDocument();
    expect(line2).toBeInTheDocument();
  });

  test("Should match snapshot", () => {
    const tree = renderer.create(<NoStocksSelected />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
