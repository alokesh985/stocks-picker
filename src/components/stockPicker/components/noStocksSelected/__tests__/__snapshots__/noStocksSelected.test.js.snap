// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the NoStocksSelected component Should match snapshot 1`] = `
<div
  className="container"
>
  <h2>
    No Stocks Selected!
  </h2>
  <h2>
    Please search for a stock on the search bar above
  </h2>
</div>
`;
